map_view/show-location-history/src/main/java/com/esri/samples/show_location_history/ShowLocationHistorySample.java

```diff
@@ -0,0 +1,201 @@
+ * limitations under the License.
+ *
+ */
+
+package com.esri.samples.show_location_history;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Calendar;
+
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Button;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.Polyline;
+import com.esri.arcgisruntime.geometry.PolylineBuilder;
+import com.esri.arcgisruntime.geometry.SpatialReference;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.location.SimulatedLocationDataSource;
+import com.esri.arcgisruntime.location.SimulationParameters;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.LocationDisplay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol;
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+
+import org.apache.commons.io.IOUtils;
+
+public class ShowLocationHistorySample extends Application {
+
+  private MapView mapView;
+  private boolean isTrackingEnabled = false;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Show Location History Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map with a dark gray canvas basemap
+      ArcGISMap map = new ArcGISMap("https://www.arcgis.com/home/item.html?id=1970c1995b8f44749f4b9b6e81b5ba45");
+
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // set the map views's viewpoint centered on Los Angeles, California and scaled
+      Point center = new Point(-13185535.98, 4037766.28, SpatialReference.create(3857));
+      mapView.setViewpoint(new Viewpoint(center, 7000));
+
+      // create a button that toggles the location tracking
+      Button trackingButton = new Button("Start tracking");
+      trackingButton.setDisable(true);
+
+      // enable the button interactions when the map is loaded
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          trackingButton.setDisable(false);
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
+        }
+      });
```

Comment:
I'd suggest perhaps moving this beneath the creation of the GraphicsOverlays and Polyline Builder. That way we have all the parts up and ready before want to make use of them inside the map. 

Then, perhaps we should move the parts that set up the `SimulatedLocationDataSource` and configure the `LocationDisplay` (lines 122-154) inside this load-done check. After all, there's not much of a reason to start the LocationDisplay if our map is not working.